src/BigQuery/Table.php

```diff
@@ -0,0 +1,337 @@
+            ],
+            $options
+        );
+
+        $response = $this->connection->insertJob($config);
+
+        return new Job($this->connection, $response['jobReference']['jobId'], $this->identity['projectId'], $response);
+    }
+
+    /**
+     * Runs an extract job which exports the contents of a table to Cloud
+     * Storage.
+     *
+     * Example:
+     * ```
+     * $destinationObject = $storage->bucket('myBucket')->object('tableOutput');
+     * $table->extract($destinationObject);
+     * ```
+     *
+     * @see https://cloud.google.com/bigquery/docs/reference/v2/jobs Jobs insert API Documentation.
+     *
+     * @param Object $destination The destination object.
+     * @param array $options {
+     *     Configuration options.
+     *
+     *     @type array $jobConfig Configuration settings for an extract job are
+     *           outlined in the [API Docs for `configuration.extract`](https://goo.gl/SQ9XAE).
+     *           If not provided default settings will be used.
+     * }
+     * @return Job
+     */
+    public function extract(Object $destination, array $options = [])
+    {
+        $objIdentity = $destination->getIdentity();
+        $config = $this->buildJobConfig(
+            'extract',
+            $this->identity['projectId'],
+            [
+                'sourceTable' => $this->identity,
+                'destinationUris' => ['gs://' . $objIdentity['bucket'] . '/' . $objIdentity['object']]
+            ],
+            $options
+        );
+
+        $response = $this->connection->insertJob($config);
+
+        return new Job($this->connection, $response['jobReference']['jobId'], $this->identity['projectId'], $response);
+    }
+
+    /**
+     * Runs a load job which loads the provided data into the table.
+     *
+     * Example:
+     * ```
+     * $table->load(fopen('/path/to/my/data.csv', 'r'));
+     * ```
+     *
+     * @see https://cloud.google.com/bigquery/docs/reference/v2/jobs Jobs insert API Documentation.
+     *
+     * @param string|resource|StreamInterface $data The data to load.
+     * @param array $options {
+     *     Configuration options.
+     *
+     *     @type array $jobConfig Configuration settings for a load job are
+     *           outlined in the [API Docs for `configuration.load`](https://goo.gl/j6jyHv).
+     *           If not provided default settings will be used.
+     * }
+     * @return Job
+     */
+    public function load($data, array $options = [])
+    {
+        $config = $this->buildJobConfig(
+            'load',
+            $this->identity['projectId'],
+            ['destinationTable' => $this->identity],
+            $options
+        );
+
+        if (!$data) {
+            return $this->connection->insertJob($config);
+        }
+
+        $config['data'] = $data;
+
+        return $this->connection->insertJobUpload($config)->upload();
+    }
+
+    /**
+     * Runs a load job which loads data from a file storage in a storage
```

Comment:
I think you were going for something like: "Runs a load job which loads data from a file in a Storage bucket..."